examples/camera-app/linux/src/clusters/camera-avstream-mgmt/camera-av-stream-manager.cpp

```diff
@@ -49,20 +49,23 @@ Protocols::InteractionModel::Status CameraAVStreamManager::VideoStreamAllocate(c
 
     for (VideoStream & stream : mCameraDeviceHAL->GetAvailableVideoStreams())
     {
-        if (!stream.isAllocated)
+        if (stream.IsCompatible(allocateArgs))
         {
             foundAvailableStream = true;
 
-            if (stream.IsCompatible(allocateArgs))
+            outStreamID = stream.videoStreamParams.videoStreamID;
+            if (!stream.isAllocated)
             {
                 stream.isAllocated = true;
-                outStreamID        = stream.videoStreamParams.videoStreamID;
 
                 // Start the video stream from HAL for serving.
                 mCameraDeviceHAL->StartVideoStream(outStreamID);
-
-                return Status::Success;
             }
+            else
+            {
+                ChipLogProgress(Camera, "Matching pre-allocated stream with ID: %d exists", outStreamID);
```

Comment:
Where does the spec allow doing this?  It says a new stream needs to be allocated, no?

Reply:
Under "Stream sharing and reuse" which states that the camera SHOULD attempt to reuse an existing stream (if it matches).

Reply:
Well, that SHOULD explicitly contradicts SHALL-level requirements in the actual command description:

> On receipt of this command, the camera SHALL allocate resources for a video stream with the given parameters and generate a unique stream identifier.

and in "VideoStreamAllocateResponse":

> This field SHALL be a VideoStreamIDType representing the newly created unique video stream identifier.

Reply:
Yes the intent is to re-use, potentially. We can adjust the wording in the Allocate command handling to match the "Stream sharing and reuse section". However, I think we can keep the name as StreamAllocate and add required text to explain that subsequent allocation attempts may use already allocated streams.

Reply:
The name is not normative, so naming it whatever is fine.  The issue is the set of normative requirements.

The place where I suggested replacing "allocate" with "ensure" is in the normative requirement, not in the command name.

Reply:
Posted Spec PR: https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/11618

Reply:
Spec PR has been merged